### DISCUSSION TITLE
Plots should not be made based on the private key

### DISCUSSION THREAD

**Discussion by i00** [Chia-Network/chia-blockchain#4584] [Ideas]
Plots should be made based on a public key not a private one:
Case:
I have a machine that I mine on
I want another set of machines that I want to use to build plots
I do not want to give the set that builds them access to the key

This does not seem possible; if the plots were based on the public key of the plot, you could have machines build plots for you without having access to your wallet

**Comment by SackofFluor** [comment]
there is a whole section in the FAQ on how EXACTLY this problem is resolved..

**Reply by i00** [reply]
I assume you mean making a "throw away key"; then use that to create plots, and add the key to the farmer box?

I am against this (in its current state now I think about it) mainly because:
- you can't alias your keys in the interface
- there is no way to view what key a plot corresponds with in plots (in filename would be nice too)

**Comment by lvcivs** [comment]
Only the “Farmer Public Key” (farmer_pk) and the “Pool Public Key” (pool_pk) are needed for plotting. So you never need to give out a private key to have plots made for you by another person / on another machine. You use `chia plots create` from the CLI with the `-f` and the `-p` option. This is all explained on the wiki.
